src/proxy_configuration.js

```diff
@@ -348,7 +349,9 @@ export class ProxyConfiguration {
     async _checkAccess() {
         const status = await this._fetchStatus();
         if (status) {
-            const { connected, connectionError } = status;
+            const { connected, connectionError, hasRestrictedResidentialProxy } = status;
+            this.hasRestrictedResidentialProxy = hasRestrictedResidentialProxy;
```

Comment:
Maybe use `!!` to give this a type in every case?

Reply:
This is not necessary, `hasRestrictedResidentialProxy` is a boolean.